escrow/state_service.go

```diff
@@ -51,6 +51,29 @@ func (service *PaymentChannelStateService) GetChannelState(context context.Conte
 		return nil, errors.New("only channel signer can get latest channel state")
 	}
 
+	// check if nonce matches with blockchain or not
+	nonceEqual, err := service.channelService.StorageNonceMatchesWithBlockchainNonce(&PaymentChannelKey{ID: channelID})
+	if err != nil {
+		log.WithError(err).Infof("payment data not available in payment storage.")
+	} else if !nonceEqual {
```

Comment:
It is worth to add unit tests for this logic.

Reply:
Added

Reply:
I see that negative cases like "storage error", "channel error" are not covered. Is it big amount of work to cover them?

Reply:
I will take a shot at this and let you know if I need anything for help

Reply:
Apologies missed updating here , this is done and also fixed a minor bug noticed ( we were just logging and not returning the error)